server/forms/GovukFrontendDateInput.ts

```diff
@@ -9,47 +9,67 @@ interface GovUkFrontendDateInputItem {
 }
 
 export default class GovukFrontendDateInput {
-  items: GovUkFrontendDateInputItem[]
-
-  private dayValue: string
-
-  private monthValue: string
-
-  private yearValue: string
-
-  constructor(query: ParsedQs, key: string, hasError: boolean = false) {
-    this.dayValue = GovukFrontendDateInput.getDatePartQueryValue(query, key, 'day')
-    this.monthValue = GovukFrontendDateInput.getDatePartQueryValue(query, key, 'month')
-    this.yearValue = GovukFrontendDateInput.getDatePartQueryValue(query, key, 'year')
-
-    this.buildItems(hasError)
-  }
-
-  private buildItems(hasError: boolean): void {
+  static getDateItems<K extends string>(
+    dateInputObj: ObjectWithDateParts<K>,
+    key: K,
+    hasError: boolean = false,
+  ): GovUkFrontendDateInputItem[] {
     const errorClass = hasError ? ' govuk-input--error' : ''
 
-    this.items = [
+    return [
       {
         name: 'day',
         classes: `govuk-input--width-2${errorClass}`,
-        value: this.dayValue,
+        value: dateInputObj[`${key}-day`] ? (dateInputObj[`${key}-day`] as string) : '',
       },
       {
         name: 'month',
         classes: `govuk-input--width-2${errorClass}`,
-        value: this.monthValue,
+        value: dateInputObj[`${key}-day`] ? (dateInputObj[`${key}-month`] as string) : '',
       },
       {
         name: 'year',
         classes: `govuk-input--width-4${errorClass}`,
-        value: this.yearValue,
+        value: dateInputObj[`${key}-day`] ? (dateInputObj[`${key}-year`] as string) : '',
       },
     ]
   }
 
-  static getDatePartQueryValue(query: ParsedQs, key: string, type: GovukFrontendDateInputPartType): string {
-    const value = query[`${key}-${type}`]
+  static dateIsComplete<K extends string>(dateInputObj: ObjectWithDateParts<K>, key: K) {
+    const dateParts = ['year', 'month', 'day'] as const
+
+    let allPartsExist = true
+
+    for (const part of dateParts) {
```

Comment:
Would `const allPartsExist = dateParts.every(part => doCheck)` work here? Find the default true a bit odd

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every